internal/cli/artifact_list.go

```diff
@@ -62,6 +62,9 @@ func (c *ArtifactListCommand) Run(args []string) int {
 		}
 		sort.Sort(serversort.ArtifactStartDesc(resp.Artifacts))
 
+		// Output the app's name before the json or table data
+		app.UI.Output("%s", app.Ref().Application, terminal.WithHeaderStyle())
+
```

Comment:
Hm, leaving a comment here since this is the command I tried, but I think the `WithMultipleApp` CLI option needs some tweaking. Before this change I was able to list artifacts for remote projects, but now I can't:

```
brian@localghost:waypoint λ waypoint artifact list go-gitops-0/go                                                                                                                                                                                                                                                   ±[PROJECT-OPS]
! This command requires a targeted project.
brian@localghost:waypoint λ waypoint artifact list -project=go-gitops-0                                                                                                                                                                                                                                             ±[PROJECT-OPS]
! failed to load config: A Waypoint configuration file (waypoint.hcl) is required but wasn't found.
  The command can use configuration stored on the server by adding project/app
  as a final argument on the command line.
  For instance: `waypoint deployment list webapp/web` for the app 'web' in
  the project 'webapp'
brian@localghost:waypoint λ waypoint artifact list -project=go-gitops-0 -app=go                                                                                                                                                                                                                                     ±[PROJECT-OPS]
! failed to load config: A Waypoint configuration file (waypoint.hcl) is required but wasn't found.
  The command can use configuration stored on the server by adding project/app
  as a final argument on the command line.
  For instance: `waypoint deployment list webapp/web` for the app 'web' in
  the project 'webapp'
```

Before this change:

```
brian@localghost:waypoint λ waypoint artifact list go-gitops-0/go                                                                                                                                                                                                                                                   ±[PROJECT-OPS]
    | ID | REGISTRY | DETAILS |    STARTED     |   COMPLETED
----+----+----------+---------+----------------+-----------------
  ✔ |  1 | docker   | build:1 | 27 minutes ago | 27 minutes ago
```

Reply:
This is true for remote `waypoint up`s too.

```
brian@localghost:waypoint λ waypoint up -remote go-gitops-0/go                                                                                                                                                                                                                                                      ±[PROJECT-OPS]
! This command requires a targeted project.
brian@localghost:waypoint λ waypoint up -remote -project=go-gitops-0 -app=go                                                                                                                                                                                                                                        ±[PROJECT-OPS]
! failed to load config: A Waypoint configuration file (waypoint.hcl) is required but wasn't found.
  The command can use configuration stored on the server by adding project/app
  as a final argument on the command line.
  For instance: `waypoint deployment list webapp/web` for the app 'web' in
  the project 'webapp'
brian@localghost:waypoint λ waypoint up -remote -project=go-gitops-0                                                                                                                                                                                                                                                ±[PROJECT-OPS]
! failed to load config: A Waypoint configuration file (waypoint.hcl) is required but wasn't found.
  The command can use configuration stored on the server by adding project/app
  as a final argument on the command line.
  For instance: `waypoint deployment list webapp/web` for the app 'web' in
  the project 'webapp'
```